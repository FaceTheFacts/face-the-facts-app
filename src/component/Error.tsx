import React from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import error from '../error_logo.png';
import {ReloadIcon, LogoIcon} from '../icons';
import {Colors} from '../theme';
import Icon from './Icon';
import RNRestart from 'react-native-restart';

const ErrorCard = () => {
  return (
    <View style={styles.container}>
      <View style={styles.logoContainer}>
        <Icon style={styles.logoIcon} icon={LogoIcon} />
        <Image style={styles.logoError} source={error} resizeMode={'contain'} />
      </View>
      <View style={styles.textContainer}>
        <Text style={styles.title}>Ups, hier ist etwas schief gegangen.</Text>
        <Text style={styles.subtitle}>
          Stelle sicher, dass du über eine aktive Internetverbindung verfügst
          und versuche es später noch einmal.
        </Text>
        <TouchableOpacity
          style={styles.linkBtn}
          onPress={() => RNRestart.Restart()}>
          <Icon style={styles.icon} icon={ReloadIcon} />
          <Text style={styles.title}>Neu laden</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: Colors.background,
    justifyContent: 'center',
    flex: 1,
    margin: 28,
    //width,
    //height,
  },
  logoContainer: {
    flex: 1,
    flexDirection: 'row',
    marginTop: '70%',
    alignItems: 'center',
    marginBottom: -10,
  },
  logoIcon: {
    flex: 1,
    color: Colors.foreground,
    height: 56,
    width: 44,
    marginLeft: 45,
    marginRight: -32,
  },
  logoError: {
    flex: 1,
    height: 18,
    width: 166,
  },
  textContainer: {
    flex: 1,
    margin: 0,
  },
  title: {
    fontFamily: 'Inter',
    fontWeight: '600',
    fontSize: 13,
    color: Colors.foreground,
  },
  subtitle: {
    fontFamily: 'Inter',
    fontSize: 11,
    color: Colors.white70,
    marginVertical: 8,
  },
  linkBtn: {
    flexDirection: 'row',
    paddingVertical: 8,
    paddingHorizontal: 12,
    marginTop: 8,
    width: 110,
    backgroundColor: Colors.background,
    borderRadius: 4,
    borderColor: Colors.white40,
    borderWidth: 1,
  },
  icon: {
    height: 13,
    width: 13,
    color: Colors.foreground,
    marginTop: 2,
    marginRight: 8,
  },
});

export default ErrorCard;
