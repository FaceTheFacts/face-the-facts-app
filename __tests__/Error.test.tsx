import 'react-native';
import renderer from 'react-test-renderer';
import Error from '../src/component/Error';

it('Error component renders correctly', () => {
  const tree = renderer.create(Error()).toJSON();
  expect(tree).toMatchSnapshot();
});
