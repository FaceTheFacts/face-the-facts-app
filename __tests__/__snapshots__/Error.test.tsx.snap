// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#181924",
      "flex": 1,
      "justifyContent": "center",
      "margin": 28,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": -10,
        "marginTop": "70%",
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={56}
      bbWidth={44}
      color={4294967295}
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "color": "#ffffff",
            "flex": 1,
            "height": 56,
            "marginLeft": 45,
            "marginRight": -32,
            "width": 44,
          },
          Object {
            "flex": 0,
            "height": 56,
            "width": 44,
          },
        ]
      }
      tintColor={4294967295}
      vbHeight={56}
      vbWidth={44}
    >
      <RNSVGGroup>
        <RNSVGPath
          d="M42.786 27.2304C42.5967 27.2034 42.4057 27.191 42.2145 27.1931C40.4118 27.1931 38.6084 27.1931 36.8043 27.1931H36.0733V26.727C36.0733 25.1463 36.1015 23.5635 36.0733 21.9849C36.0497 20.7509 35.6389 19.5576 34.9018 18.5821C34.1648 17.6065 33.1408 16.9008 31.9818 16.5695C31.3558 16.3996 30.6954 16.3624 29.9623 16.2567C30.0543 16.1991 30.142 16.1347 30.2249 16.064C32.5513 13.6609 33.5651 10.804 33.0016 7.43128C32.6164 5.14895 31.3845 3.10834 29.5644 1.73755C27.7444 0.366751 25.4775 -0.227712 23.2395 0.0788547C18.5947 0.700362 15.099 4.95769 15.6765 10.3917C15.8785 12.3702 16.7529 14.0483 18.0413 15.513C18.2736 15.7761 18.5159 16.0329 18.7744 16.3147C17.5627 16.4825 16.4783 16.8513 15.5433 17.6012C14.0084 18.8277 13.3602 20.4912 13.34 22.4448C13.3259 23.895 13.34 25.3576 13.34 26.8161V27.1517H10.1088C10.0978 27.0956 10.0904 27.0389 10.0866 26.9818C10.0644 24.5952 10.0543 22.1569 10.018 19.7703C10.0058 18.9623 10.2906 18.403 10.7833 17.7815C11.2862 17.1497 11.7348 16.4746 12.1243 15.7636C12.6675 14.7568 11.9445 13.6443 10.8177 13.6629C10.1714 13.6733 9.7312 14.0524 9.37981 14.562C8.87494 15.2954 8.40036 16.0516 7.87327 16.7684C7.31128 17.5204 7.01059 18.4436 7.01904 19.3912C7.01096 21.8606 7.00624 24.3301 7.0049 26.7995V27.1931H6.61514C4.96321 27.1931 3.3106 27.1931 1.65732 27.1931C1.45419 27.1921 1.2514 27.2101 1.05148 27.247C0.756311 27.2812 0.484469 27.428 0.289928 27.6583C0.0953878 27.8885 -0.00762327 28.1854 0.00134792 28.49C-0.0064075 28.8103 0.0193205 29.1304 0.0780891 29.4451C0.805101 33.3979 1.53548 37.3513 2.26922 41.3055C2.47117 42.3828 2.90738 42.7557 3.9777 42.8075C4.07868 42.8075 4.17965 42.8075 4.2786 42.8075H8.77396V56H11.5629V36.4681H32.4705V55.9979H35.1544V42.7743H35.5442C37.0144 42.7743 38.4865 42.7743 39.9567 42.7743C41.021 42.7743 41.5117 42.3268 41.7076 41.2516C42.0752 39.2338 42.4589 37.2181 42.8284 35.2002C43.2141 33.1016 43.5999 31.0009 43.9674 28.9002C44.129 27.9742 43.6827 27.363 42.786 27.2304ZM18.4877 9.0617C18.501 7.43107 19.1441 5.87243 20.2761 4.72754C21.408 3.58265 22.9363 2.94498 24.5259 2.95436C27.6763 2.98337 30.3218 5.8423 30.2875 9.16944C30.2511 12.6437 27.6197 15.3638 24.326 15.3306C21.0322 15.2975 18.4533 12.5359 18.4877 9.0617ZM16.3127 21.699C16.4056 20.4166 17.4093 19.5237 18.6613 19.3787C18.9114 19.3484 19.1628 19.3325 19.4146 19.3311C23.0113 19.3311 26.608 19.3311 30.2047 19.3311C31.4164 19.3311 32.4019 19.7848 32.9209 20.9698C33.097 21.3747 33.1944 21.811 33.2076 22.2542C33.242 23.7313 33.2218 25.2105 33.2218 26.6877C33.2218 26.7374 33.2218 26.7871 33.2076 26.8575H16.1895C16.2238 25.1173 16.1895 23.3999 16.3147 21.6949L16.3127 21.699ZM39.365 39.5819C39.2661 40.1288 39.151 40.2303 38.6279 40.2324C37.5798 40.2324 36.3318 40.2324 35.2352 40.2324V39.7766V39.441C35.2352 38.2228 35.2352 37.0067 35.2352 35.7886C35.2358 35.5496 35.2004 35.312 35.1302 35.0842C34.8838 34.2949 34.2133 33.8018 33.3126 33.8018H10.6379C9.45049 33.8018 8.9335 34.1333 8.79416 35.0863C8.76422 35.3003 8.75005 35.5164 8.75175 35.7327V40.1847C8.6326 40.1847 8.52557 40.1992 8.41853 40.1992C7.36235 40.1992 6.10219 40.1992 5.04399 40.1992C4.79155 40.1992 4.70673 40.1288 4.6623 39.8657C4.13118 36.7581 3.10932 30.4788 3.02249 29.9361H40.9705L39.365 39.5819Z"
          fill={
            Array [
              2,
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/error_logo.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 18,
          "width": 166,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "margin": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#ffffff",
          "fontFamily": "Inter",
          "fontSize": 13,
          "fontWeight": "600",
        }
      }
    >
      Ups, hier ist etwas schief gegangen.
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(252 ,252 ,252, 0.7)",
          "fontFamily": "Inter",
          "fontSize": 11,
          "marginVertical": 8,
        }
      }
    >
      Stelle sicher, dass du über eine aktive Internetverbindung verfügst und versuche es später noch einmal.
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#181924",
          "borderColor": "rgba(252 ,252 ,252, 0.4)",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginTop": 8,
          "opacity": 1,
          "paddingHorizontal": 12,
          "paddingVertical": 8,
          "width": 110,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={13}
        bbWidth={13}
        color={4294967295}
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "color": "#ffffff",
              "height": 13,
              "marginRight": 8,
              "marginTop": 2,
              "width": 13,
            },
            Object {
              "flex": 0,
              "height": 13,
              "width": 13,
            },
          ]
        }
        tintColor={4294967295}
        vbHeight={12}
        vbWidth={13}
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M12.5938 0.84375V4.5C12.5938 4.95703 12.2129 5.3125 11.7812 5.3125H8.125C7.66797 5.3125 7.3125 4.95703 7.3125 4.5C7.3125 4.06836 7.66797 3.6875 8.125 3.6875H9.72461C8.96289 2.69727 7.76953 2.08789 6.47461 2.08789C4.24023 2.08789 2.4375 3.89062 2.4375 6.125C2.4375 8.38477 4.24023 10.1875 6.47461 10.1875C7.36328 10.1875 8.20117 9.9082 8.91211 9.375C9.26758 9.12109 9.77539 9.19727 10.0547 9.55273C10.334 9.9082 10.2578 10.416 9.90234 10.6953C8.91211 11.4316 7.71875 11.8125 6.47461 11.8125C3.35156 11.8125 0.8125 9.27344 0.8125 6.125C0.8125 3.00195 3.35156 0.462891 6.47461 0.462891C8.25195 0.462891 9.87695 1.27539 10.9688 2.64648V0.84375C10.9688 0.412109 11.3242 0.03125 11.7812 0.03125C12.2129 0.03125 12.5938 0.412109 12.5938 0.84375Z"
            fill={
              Array [
                2,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Inter",
            "fontSize": 13,
            "fontWeight": "600",
          }
        }
      >
        Neu laden
      </Text>
    </View>
  </View>
</View>
`;
